users/permissions.py

```diff
@@ -0,0 +1,19 @@
+from rest_framework.permissions import BasePermission
+
+SAFE_METHODS = ['GET', 'HEAD', 'OPTIONS']
+
+
+class IsAuthenticatedAndOwner(BasePermission):
+    message = 'Permission denied.'
+
+    def has_permission(self, request, view):
+        if request.method in SAFE_METHODS:
+            return True
+        else:
+            return request.user and request.user.is_authenticated
+
+    def has_object_permission(self, request, view, obj):
+        if request.method in SAFE_METHODS:
+            return True
+        else:
+            return obj == request.user
```

Comment:
Permission name and how the permission class works is inconsistent. You are letting other users to access the object with GET method. But the permission class says `only owners can access`. 

Let's remove allowing safe methods.

If allowing safe method is needed you can use bitwise operations for combination of permission classes.

For example: permission_classes = (IsOwner | ReadOnly,) 

Please see https://www.django-rest-framework.org/api-guide/permissions/#setting-the-permission-policy